dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -186,7 +186,7 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
             try
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                context = await f.InvokeAsync(context).ConfigureAwait(false);
+                context = await f.InvokeAsync(context, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
We shouldn't need to make this change since the token is in the `context`, right? Just want to make sure users don't have to change behavior if they are already calling `InvokeAsync` themselves with `context`. Fine with the change here though. Does make me wonder if we should then do similar with `this._log` though.

Reply:
If it's the same token, yes.  This gets at the issue I mentioned in the description above:
> One issue I noticed I haven't addressed here: Some APIs take both an SKContext and a CancellationToken separately. If both are supplied, one of the tokens (either the explicitly passed one or the one from the SKContext) might be ignored. If they're not the same token and they're both cancelable, a combined token should really be passed on to the target so that the target ends up monitoring both.

The API is designed in such a way that we're potentially now taking in two different tokens and we might end up ignoring one of them.  Ideally that wouldn't be possible by construction, e.g. if SKContext is the way we want to pass around a contained token, then a) APIs taking SKContext shouldn't also take a CancellationToken and b) it should be possible to easily create a new SKContext that contains a CancellationToken the dev wants to use.  Alternatively, we need to ensure that if there are two different tokens, that all of them which are cancelable are respected (I haven't done that here).

Reply:
By design we should assume that the context has the right cancellation token (managed by the kernel) and that there's no need to pass another one.

Reply:
considering that `context` is created locally using the `cancellationToken` passed in to the function, I would use `context.CancellationToken` to avoid the ambiguity.